examples/plugins/adder/main.go

```diff
@@ -15,45 +15,40 @@ import (
 //
 // CONSTRAINT: All fields of the config struct need to be exportable
 type config struct {
-	Key5 string
-	Key6 string
-	Key7 string
-	Key8 string
+	LocConfig localize.Config
 }
 
 // Initializes a zero-valued struct and stores its address
 var cfg = &config{}
 
 // AdderCommand returns the root command of plugin.
 // This will be added to the host CLI as an extension.
-func AdderCommand(h *c.CfgHandler) (*cobra.Command, error) {
-	// Sets dummy values into config
-	cfg.Key5 = "val5"
-	cfg.Key6 = "val6"
-	cfg.Key7 = "val7"
-	cfg.Key8 = "val8"
+func AdderCommand(f *factory.Factory) (*cobra.Command, error) {
 
 	// Stores the config for localizer
-	locConfig := localize.Config{
+	cfg.LocConfig = localize.Config{
 		Language: language.English,
 		Path:     "examples/plugins/adder/locales/en/adder.en.yaml",
 		Format:   "yaml",
 	}
 
 	// Initializes the localizer by passing config
-	loc, err := localize.InitLocalizer(locConfig)
+	loc, err := localize.InitLocalizer(cfg.LocConfig)
 	if err != nil {
 		return nil, err
 	}
 
-	//Stores a new factory instance with default settings
-	f := factory.Default(loc)
+	opts := &factory.Factory{
+		Logger:     f.Logger,
+		Localizer:  loc,
+		CfgHandler: f.CfgHandler,
+	}
```

Comment:
I can see why we do this. We need the logger and config from the host factory but we need to setup a localizer based on the plugin. I just find this weird looking but honestly not sure what else to do besides this either.

Reply:
Agreed.
This is just a temporary workaround to get things going for the time being. But we need to think of a better way to handle this in future. 

I'll create a separate issue for discussion on this, tomorrow.